nomad/structs/csi.go

```diff
@@ -152,7 +152,9 @@ func (o *CSIMountOptions) Copy() *CSIMountOptions {
 	if o == nil {
 		return nil
 	}
-	return &(*o)
+
+	no := *o
+	return &no
```

Comment:
`CSIMountOptions` contains `MountFlags []string` - I think we need to also copy the slice, e.g. with `helper.CopySliceString`

Reply:
Good catch - thanks!